### ISSUE TITLE
Pen stops working after resume

### ISSUE THREAD

**Issue by lviggiani** [linux-surface/iptsd#37]
This seems to be a regression... I remember that in the past this issue was there but was also fixed. However since a couple of days (kernel upgrade?) the issue is back. After resuming from sleep, the pen won't work anymore and I have to restart the device.
SP4 ArchLinux 

- 5.10.16-arch1-3-surface
- all up to date packages

**Comment by StollD** [comment]
Please post your dmesg and journalctrl logs

**Comment by lviggiani** [comment]
here are ```dmesg``` and ```journalctl -e``` outputs
[dmesg.log](https://github.com/linux-surface/iptsd/files/6038970/dmesg.log)
[jourlanctl.log](https://github.com/linux-surface/iptsd/files/6038971/jourlanctl.log)

**Comment by StollD** [comment]
What is `ipts-sleep.service`?

**Comment by lviggiani** [comment]
It's a workaround to the sleep problem that was working in the past and that I got from here:

https://github.com/linux-surface/linux-surface/issues/89#issuecomment-595139027

It was doing the job until a couple of days ago.
BTW, I have stopped / disabled it but I still have the same issue (pen not working after resume).

**Comment by StollD** [comment]
Does it work if you downgrade the kernel to 5.10.13?

**Comment by StollD** [comment]
Yes, please test with kernel 5.10.13. Also, it would be great if you could try this branch of the ipts module on 5.10.13: https://github.com/linux-surface/intel-precise-touch/commits/suspend-fun

**Comment by lviggiani** [comment]
I've tried kernel 5.10.13, standard ipts module and **without** ```ipts-sleep.service``` and **it works**

```
luca@mssp4luca ~> uname -r
5.10.13-arch1-1-surface
luca@mssp4luca ~> systemctl status ipts-sleep.service
● ipts-sleep.service - IPTS sleep hook
     Loaded: loaded (/etc/systemd/system/ipts-sleep.service; disabled; vendor p>
     Active: inactive (dead)
```

So ```ipts-sleep.service``` seems not be needed anymore (it was in the past).
I'll try the ```suspend-fun``` and let you know. Do you also want me to test it on 5.10.13 and / or 5.10.16 ?

**Comment by StollD** [comment]
Yes, please test it on 5.10.13.

**Comment by lviggiani** [comment]
Hi sorry for my late reply but I've been busy during the last days.
Just to make sure:

```
git clone https://github.com/linux-surface/intel-precise-touch.git
git checkout suspend-fun
cd intel-precise-touch
make all
sudo insmod ipts.ko
```

In build instructions, when you say "Make sure you applied the patches from patches/ to your kernel", I can assume they are as I'm sunning linux-surface kernel, right?

**Comment by lviggiani** [comment]
By following the above instructions (ere they right?) suspend / resume works

**Comment by StollD** [comment]
Yes, these steps are correct, thank you for testing!

> One thing I noticed: with this new module, the pen cursor in Firefox appears a an arrow (like mouse) where as before it was shown as a crosshair like when moving the pen on desktop

Thats normal. gnome-shell draws its cursor as a crosshair, while firefox doesnt make a difference between stylus and mouse.

**Comment by lviggiani** [comment]
Hi, today I have upgraded to linux 5.10.16-arch1-3-surface and pen on resume is now working again

**Comment by lviggiani** [comment]
Hi is there any progress on this issue or do I have to stay on 5.10.13? Thaks! :)

**Comment by StollD** [comment]
Sorry, I have been kinda busy with exams the last days. The code is there but I havent worked on merging it into the kernel tree and the patches yet. I will try to get that done today.

**Comment by lviggiani** [comment]
Hi, thanks for your answer and sorry I didn't mean to put pressure!
Great, take your time!

**Comment by StollD** [comment]
This should be fixed with the latest kernel updates.

**Comment by lviggiani** [comment]
It was a false alarm. I've rebooted and tried again and it seems to be working even after 4 subsequent suspend / resumes.
Thanks!

**Comment by lviggiani** [comment]
Fixed by commit dd85e81
